quinn/src/connect.rs

```diff
@@ -0,0 +1,21 @@
+use std::{
+    io,
+    net::{IpAddr, Ipv4Addr, Ipv6Addr, SocketAddr},
+};
+
+use crate::{ClientConfig, Connecting, Endpoint};
+
+/// Connect to `server_address`, authenticating it as `server_name`, using a new endpoint
```

Comment:
Seems reasonable, though I don't think it should live in its own module. Maybe an associated method for `ClientConfig`?